services/common/rs_server_common/stac_api_common.py

```diff
@@ -503,12 +543,40 @@ def read_cql(filt: dict):
             raise first_exception
 
         # Return results as a dict
-        return stac_pydantic.ItemCollection(features=list(all_features.values()), type="FeatureCollection").model_dump()
+        data = stac_pydantic.ItemCollection(features=list(all_features.values()), type="FeatureCollection").model_dump()
+
+        # Handle pagination links.
+        data["next"] = f"page={self.page + 1}"
+        if self.page > 1:
+            data["prev"] = f"page={self.page - 1}"
+
+        return data
 
     @abstractmethod
     async def process_search(self, collection: dict, odata_params: dict) -> stac_pydantic.ItemCollection:
         """Do the search for the given collection and OData parameters."""
 
+    def get_page(self) -> int:
+        """Get page value (integer) according to given priority"""
+        try:
+            user_page = int(self.request.query_params.get("page", self.odata.get("page", 1)))  # type: ignore
```

Comment:
This doesn't work, `self.odata.get("page")` doesn't exit. You should use `self.page` instead.

Reply:
Agree